src/main/java/net/imagej/ops/filter/tubeness/DefaultTubenessOp.java

```diff
@@ -0,0 +1,263 @@
+		final Img< DoubleType > gradient = factory.create( gradientDimensions, new DoubleType() );
+		final Img< DoubleType > gaussian = factory.create( input, new DoubleType() );
+		
+		// Handle multithreading.
+		final int nThreads = Runtime.getRuntime().availableProcessors();
+		final ExecutorService es = threadService.getExecutorService();
+
+		try
+		{
+			// Hessian calculation.
+			HessianMatrix.calculateMatrix(
+					Views.extendBorder( input ),
+					gaussian,
+					gradient,
+					hessian,
+					new OutOfBoundsBorderFactory<>(),
+					nThreads, es, sigma );
+
+			statusService.showProgress( 1, 3 );
+			if ( isCanceled() )
+				return;
+
+			// Hessian eigenvalues.
+			final RandomAccessibleInterval< DoubleType > evs = TensorEigenValues.calculateEigenValuesSymmetric(
+					hessian,
+					TensorEigenValues.createAppropriateResultImg( hessian, factory, new DoubleType() ),
+					nThreads, es );
+
+			statusService.showProgress( 2, 3 );
+			if ( isCanceled() )
+				return;
+
+			final AbstractUnaryComputerOp< Iterable< DoubleType >, DoubleType > method;
+			switch ( numDimensions )
+			{
+			case 2:
+				method = new Tubeness2D( sigma );
+				break;
+			case 3:
+				method = new Tubeness3D( sigma );
+			default:
+				System.err.println( "Cannot compute tubeness for " + numDimensions + "D images." );
+				return;
+			}
+			ops().transform().project( tubeness, evs, method, numDimensions );
+
+			statusService.showProgress( 3, 3 );
+
+			return;
+		}
+		catch ( final IncompatibleTypeException | InterruptedException | ExecutionException e )
+		{
+			e.printStackTrace();
+			return;
+		}
+	}
+
+	private static final class Tubeness2D extends AbstractUnaryComputerOp< Iterable< DoubleType >, DoubleType >
+	{
+
+		private double sigma;
+
+		public Tubeness2D( final double sigma )
+		{
+			this.sigma = sigma;
+		}
+
+		@Override
+		public void compute( final Iterable< DoubleType > input, final DoubleType output )
+		{
+			// Use just the largest one.
+			final Iterator< DoubleType > it = input.iterator();
+			it.next();
+			final double val = it.next().get();
+			if ( val >= 0. )
+				output.setZero();
+			else
+				output.set( sigma * sigma * Math.abs( val ) );
+
+		}
+	}
+
+	private static final class Tubeness3D extends AbstractUnaryComputerOp< Iterable< DoubleType >, DoubleType >
+	{
+
+		private double sigma;
+
+		public Tubeness3D( final double sigma )
+		{
+			this.sigma = sigma;
+		}
+
+		@Override
+		public void compute( final Iterable< DoubleType > input, final DoubleType output )
+		{
+			// Use the two largest ones.
+			final Iterator< DoubleType > it = input.iterator();
+			it.next();
+			final double val1 = it.next().get();
+			final double val2 = it.next().get();
+			if ( val1 >= 0. || val2 >= 0. )
+				output.setZero();
+			else
+				output.set( sigma * sigma * Math.sqrt( val1 * val2 ) );
+
+		}
+	}
+
+	// -- Cancelable methods --
+
+	/** Reason for cancelation, or null if not canceled. */
+	private String cancelReason;
```

Comment:
How about moving this field definition to the top of the class? If you (and/or others) disagree, we can also disable the codacy rule to avoid these annoying commit quality checks :)

Reply:
Personally I prefer field definitions at the top of classes, not in the middle. https://imagej.net/Coding_style#Ordering_of_code_blocks